sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,311 @@
+//            // jsonOutput
+//            // Elements are separated by commas.
+//            if(itr.hasNext()) { jsonOutput += ", "; }
+//        }
+//
+//        // Returning the resulting String representation of the JsonArray object
+//        // JSON arrays are delimited by opening and closing square brackets.
+//        return "[" + jsonOutput + "]";
+//    }
+
+    /**
+     * @return boolean of whether this JsonElement object is of type JsonArray.
+     */
+    @Override
+    public boolean isArray() { return true; }
+
+    /**
+     * @return String representation of the JsonArray. This functionality is
+     * defined within the toJson method.
+     */
+    @Override
+    public String toString() {
+        try {
+            return this.toJson();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * The build method is used when a JsonReader is passed in to the array (usually through the constructor)
+     * It will then read the JsonReader and build the JsonArray from it. If it hits a nested element, the reader
+     * is passed into the constructor of the new element, and it builds recursively.
+     * @param reader the JsonReader to use
+     * @throws IOException if the JsonReader throws an exception
+     */
+    @Override
+    public void build(JsonReader reader) throws IOException {
+
+        while(reader.currentToken() != JsonToken.END_ARRAY) {
+
+            JsonToken token = reader.nextToken();
+            //todo remove this debug line, and remove the comment out cases
+            //System.out.printf("Token: %s%n", token);
+
+
+            switch (token) {
+
+                case FIELD_NAME:
+                    //fieldName = reader.getFieldName();
+                    break;
+                case START_OBJECT:
+                    this.addElement(new JsonObject(reader));
+                    break;
+                case START_ARRAY:
+                    this.addElement(new JsonArray(reader));
+                    break;
+                case STRING:
+                    this.addElement(new JsonString(reader.getString()));
+                    break;
+                case NUMBER:
+                    this.addElement(new JsonNumber(reader.getString()));
+                    break;
+                case BOOLEAN:
+                    this.addElement(new JsonBoolean(reader.getBoolean()));
+                    break;
+                case NULL:
+                    this.addElement(new JsonNull());
+                    break;
+//                 These two cases are picked up by the overall while statement.
+                case END_DOCUMENT:
+                    return;
+                case END_ARRAY:
+                    // It shouldn't get here, the loop prevents it BUT just in case...
+                    return;
+                case END_OBJECT:
+                    //this should be an error, it's not possible
+                    break;
```

Comment:
Two things, missing a `default` case statement and I think we should remove END_DOCUMENT and END_OBJECT from the case statements and have `default` throw an error if it's reached. That way as we test, we can guarantee that deserialization doesn't have edge cases that would cause problems in production.